Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -161,97 +163,79 @@ public void Serialize<T>(Stream stream, T item)
 
         public void Serialize<T>(IIonWriter writer, T item)
         {
+            currentDepth++;
+            if (currentDepth >= options.MaxDepth)
+            {
+                currentDepth = 0;
+                throw new NotSupportedException($"Can not serialize further as max object tree depth {options.MaxDepth} is reached");

```

Comment:
Nit: Cannot is more grammatically correct

Reply:
Addressed.